microcosm_flask/paging.py

```diff
@@ -0,0 +1,89 @@
+"""
+Pagination support.
+
+"""
+from flask import request
+
+from microcosm_flask.linking import Link, Links
+from microcosm_flask.operations import Operation
+
+
+class Page(object):
+    def __init__(self, offset, limit):
+        self.offset = offset
+        self.limit = limit
+
+    @classmethod
+    def from_request(cls, default_limit=20):
+        try:
+            offset = int(request.args["offset"])
+        except:
+            offset = 0
+        try:
+            limit = int(request.args["limit"])
+        except:
+            limit = default_limit
+        return cls(offset=offset, limit=limit)
+
+    def next(self):
+        return Page(
+            offset=self.offset + self.limit,
+            limit=self.limit,
+        )
+
+    def prev(self):
+        return Page(
+            offset=self.offset - self.limit,
+            limit=self.limit,
+        )
+
+    def to_dict(self):
+        return dict(
+            offset=self.offset,
+            limit=self.limit,
+        )
+
+    def to_tuples(self):
+        """
+        Convert to tuples for deterministic order when passed to urlencode.
+
+        """
+        return [
+            ("offset", self.offset),
+            ("limit", self.limit),
+        ]
+
+
+class PaginatedList(object):
```

Comment:
I'm not using the paginated list yet; it will get fleshed out as soon as I add the CRUD layer.